### ISSUE TITLE
bug in handling of RAV4 acceleration byte-length

### ISSUE THREAD

**Issue by liljonnystyle** [jmscslgroup/strym#47]
On some (maybe all? unclear) non-hybrid RAV4's, extracting the ego acceleration results in an error. E.g., for `2021-08-02-12-18-43_2T3P1RFV2MW181087_CAN_Messages.csv`, the result is:
```
[2022_03_11_03_26_54] (root) INFO: Vehicle model infered is toyota-rav4-2021
[2022_03_11_03_26_57] (cantools.database.can.database) WARNING: Overwriting message 'GAS_PEDAL' with 'GAS_PEDAL' in the name to message dictionary.
[2022_03_11_03_27_04] (root) INFO: Found Hybrid RAV4 where acceleration messages are 4  bytes.
data not 8 bytes
```
Matt Nice says: "some of the normal RAV4's have a 4 byte acceleration message like the hybrid vehicle"

**Comment by liljonnystyle** [comment]
@MatthewNice can you provide the solution? Was it this commit? https://github.com/jmscslgroup/strym/commit/9173490f7b8f28b16e4cf24e7257ec6d1fe6751b

It seems like the issue is still unresolved, per Junqing.

**Comment by MatthewNice** [comment]
yes that's the commit. there was an issue with making the copy of the dataframe when used within the strymread class object. when I made this change, I tested the code and it worked on my machine.

**Comment by MatthewNice** [comment]
Junqing is not running the updated code, but the code on master resolves this issue when he can get it running.

**Comment by liljonnystyle** [comment]
I have been able to independently verify that the above solution works for getting accelerations, but it seems an inadvertent bug has emerged. After getting the accelerations, all other CAN messages return as empty arrays. A tentative workaround is to get all other CAN messages first before getting the accelerations.

**Comment by MatthewNice** [comment]
I have addressed this inadvertent bug in commit b55e82619be58e701b9eefed69944ff0391ed297 . The originating problem that caused this bug was the use of the CleanData function with the streamread class. The class creates a non-unique indexing for the dataframe, but the CleanData function was written under the presumption of unique indexing. The initial fix solved this problem by returning the clean data but only the acceleration message (avoiding the merge). This is bad, and creates the secondary problem @liljonnystyle describes. This new commit solves the problem correctly by fixing the issue created from streamread and adding a unique index so merging of cleaned data is possible.
